Sources/OpenAI/OpenAI.swift

```diff
@@ -127,7 +132,26 @@ extension OpenAI {
             task.resume()
         } catch {
             completion(.failure(error))
-            return
+        }
+    }
+    
+    func performSteamingRequest<ResultType: Codable>(request: any URLRequestBuildable, completion: @escaping (Result<ResultType, Error>) -> Void) {
+        do {
+            let request = try request.build(token: configuration.token, organizationIdentifier: configuration.organizationIdentifier, timeoutInterval: configuration.timeoutInterval)
+            let session = StreamingSession<ResultType>(urlRequest: request)
+            session.onReceiveContent = {_, object in
+                completion(.success(object))
+            }
+            session.onProcessingError = {_, error in
+                completion(.failure(error))
+            }
+            session.onComplete = { [weak self] object, error in
+                self?.streamingSessions.removeAll(where: { $0 == object })
```

Comment:
Heya! Jumping in, I got a little excited to test this out. Doing some testing myself, this needs to call `completion` in some way in order for the backing `AsyncStream` to actually finish. At the moment I'm throwing an error like this:
```
enum StreamError: Error {
 case streamStop
}
//...
completion(.failure(StreamError.streamStop)
```

Which allows the `throw` to act as the trigger to stop. There's likely a more elegant way to do this, so hopefully the report at least helps track that part down. Cheers, and thanks much for your time!